util/logtools/split_log.ts

```diff
@@ -1,46 +1,77 @@
 import fs from 'fs';
 import readline from 'readline';
+
+import { Namespace } from 'argparse';
+
 import Anonymizer from './anonymizer';
-import Splitter from './splitter';
+import { LogUtilArgParse, TimelineArgs } from './arg_parser';
 import { EncounterCollector, TLFuncs } from './encounter_tools';
-import ZoneId from '../../resources/zone_id';
-import argparse from 'argparse';
-import { LogUtilArgParse } from './arg_parser';
 import { ConsoleNotifier } from './notifier';
+import Splitter from './splitter';
+
+class SplitLogArgs extends Namespace implements TimelineArgs {
+  'file': string | null;
+  'force': boolean | null;
+  'search_fights': number | null;
+  'search_zones': number | null;
+  'fight_regex': string | null;
+  'zone_regex': string | null;
+  'adjust': number | null;
+}
 
 // TODO: add options for not splitting / not anonymizing.
 const timelineParse = new LogUtilArgParse();
-const args = timelineParse.parser.parseArgs();
 
-const printHelpAndExit = (errString) => {
+const args = new SplitLogArgs({});
+timelineParse.parser.parseArgs(undefined, args);
+
+const printHelpAndError = (errString: string): void => {
   console.error(errString);
   timelineParse.parser.printHelp();
-  process.exit(-1);
 };
 
-if (args.file === null && args.timeline === null)
-  printHelpAndExit('Error: Must specify at least one of -f, -t\n');
+if (args.file === null) {
+  printHelpAndError('Error: Must specify -f\n');
+  process.exit(-1);
+}
+
 let numExclusiveArgs = 0;
-for (const opt of ['search_fights', 'search_zones', 'fight_regex', 'zone_regex']) {
+const exclusiveArgs: (keyof SplitLogArgs)[] = [
+  'search_fights',
+  'search_zones',
+  'fight_regex',
+  'zone_regex',
+];
+for (const opt of exclusiveArgs) {
   if (args[opt] !== null)
     numExclusiveArgs++;
 }
-if (numExclusiveArgs !== 1)
-  printHelpAndExit('Error: Must specify exactly one of -lf, -lz, or -fr\n');
-if (args['fight_regex'] === -1)
-  printHelpAndExit('Error: -fr must specify a regex\n');
-if (args['zone_regex'] === -1)
-  printHelpAndExit('Error: -zr must specify a regex\n');
+if (numExclusiveArgs !== 1) {
+  printHelpAndError('Error: Must specify exactly one of -lf, -lz, or -fr\n');
+  process.exit(-2);
+}
 
 const logFileName = args.file;
 let exitCode = 0;
-const errorFunc = (str) => {
+const errorFunc = (str: string) => {
   console.error(str);
   exitCode = 1;
 };
 
-const writeFile = (outputFile, startLine, endLine) => {
-  return new Promise((resolve, reject) => {
+const writeFile = (
+  outputFile: string,
+  startLine: string | undefined,
+  endLine: string | undefined,
```

Comment:
It occurs to me that for the future, we could probably rewrite this to be something like (not production-ready!)
```typescript
const writeFile = (
    outputFile: string,
    collector: EncounterCollector,
    indexList: number[],
): void => {
  const lines: string[] = [];
  for (const fight of collector.fights) {
    if (indexList.includes(collector.fights.indexOf(fight)))
      lines.concat(fight.lines);
  }

  // Process lines as usual
}
```

and then set `processLine(line, true)` in our collector prepass. This would save us an additional file access when doing the actual line processing. (We might also want to add a hardcoded index to each fight/zone in `encounter_tools` as well in that case, although that might be slightly more complex.)